static/app/views/alerts/rules/metric/ruleConditionsForm.tsx

```diff
@@ -353,27 +369,139 @@ class RuleConditionsForm extends PureComponent<Props, State> {
+              isDisabled={disabled}
+              value={timeWindow}
+              onChange={({value}) => onTimeWindowChange(value)}
+              inline={false}
+              flexibleControlStateSize
+            />
+          )}
+        </FormRow>
+      </Fragment>
+    );
+  }
+
+  renderMonitorTypeSelect() {
+    const {
+      onMonitorTypeSelect,
+      monitorType,
+      activationCondition,
+      timeWindow,
+      onTimeWindowChange,
+    } = this.props;
+
+    return (
+      <Fragment>
+        <StyledListItem>
+          <StyledListTitle>
+            <div>{t('Select Monitor Type')}</div>
+          </StyledListTitle>
+        </StyledListItem>
+        <FormRow>
+          <MonitorSelect>
+            <MonitorCard
+              position="left"
+              isSelected={monitorType === MonitorType.CONTINUOUS}
+              onClick={() =>
+                onMonitorTypeSelect({
+                  monitorType: MonitorType.CONTINUOUS,
+                  activationCondition,
+                })
+              }
+            >
+              <strong>Continuous</strong>
+              <div>Continuously monitor trends for the metrics outlined below</div>
+            </MonitorCard>
+            <MonitorCard
+              position="right"
+              isSelected={monitorType === MonitorType.ACTIVATED}
+              onClick={() =>
+                onMonitorTypeSelect({
+                  monitorType: MonitorType.ACTIVATED,
+                })
+              }
+            >
+              <strong>Conditional</strong>
+              {monitorType === MonitorType.ACTIVATED ? (
+                <ActivatedAlertFields>
+                  Monitor{' '}
+                  <SelectControl
+                    name="activationCondition"
+                    styles={{
+                      control: (provided: {[x: string]: string | number | boolean}) => ({
+                        ...provided,
+                        minWidth: 200,
+                        maxWidth: 300,
+                      }),
+                      container: (provided: {
+                        [x: string]: string | number | boolean;
+                      }) => ({
+                        ...provided,
+                        margin: `${space(0.5)}`,
+                      }),
+                    }}
+                    options={[
+                      {
+                        value: ActivationConditionType.RELEASE_CREATION,
+                        label: t('New Release'),
+                      },
+                      {
+                        value: ActivationConditionType.DEPLOY_CREATION,
+                        label: t('New Deploy'),
+                      },
+                    ]}
+                    required
+                    value={activationCondition}
+                    onChange={({value}) =>
+                      onMonitorTypeSelect({activationCondition: value})
+                    }
```

Comment:
nit: prefer creating named function for these as it will redefine the function each render. if you give it a name then v8 can optimize.

Reply:
Is this true @saponifi3d? Do you have a reference that describes what V8 does here?

We've almost never use named functions for callbacks, not sure if we want to start unless this really is a thing (but I'm not so sure it is)